### PR TITLE
Fix for Issue #5855

### PR DESCRIPTION
**Pull request by daniel-rome**

### CODE REVIEW

File: libr/debug/p/debug_native.c [hunk 1]
```diff
@@ -217,6 +218,12 @@ static int r_debug_native_continue_syscall (RDebug *dbg, int pid, int num) {
 #endif
 }
 
+/* Callback to trigger SIGINT signal */
+static void r_debug_native_stop(RDebug *dbg) {
+	// I don't know if there is a better way to trigger this
+	kill (dbg->pid, SIGINT);
```
Comment (radare) [inline libr/debug/p/debug_native.c]: this is not portable. you have r_debug_kill for this

File: libr/debug/p/debug_native.c [hunk 2]
```diff
@@ -247,6 +254,7 @@ static int r_debug_native_continue (RDebug *dbg, int pid, int tid, int sig) {
 		contsig = sig;
 	}
 	//eprintf ("continuing with signal %d ...\n", contsig);
+	r_cons_break (r_debug_native_stop, dbg);
```
Comment (radare) [inline libr/debug/p/debug_native.c]: i would call r_cons_break_end() right after the ptrace call

**Comment by daniel-rome** [comment]
After add the r_cons_break_end function the handler is still working. Now, a way to avoid two SIGINT handlers within the debugging functionality should be found.

**Comment by alvarofe** [comment]
where have you been testing this? seems that in mac osx doesn't work

**Comment by Maijin** [comment]
@alvarofe unrelated

**Comment by alvarofe** [comment]
about mac os x i tested locally i didn't look at travis at all

**Comment by radare** [comment]
We need rconsbreak(push,pop) to have this working properly, as well as doing it only when target process is not in the same terminal

**Comment by alvarofe** [comment]
Mac os x is a special case. However, r_cons_break is set within r_debug_continue under #if clause affecting only linux so for the rest either XNU, BSD and such doesn't work. I have changed it in local to be made in r_debug_attach https://github.com/alvarofe/radare2/commit/b93a850319933d57de868c21274d225a01e16e1b

**Comment by alvarofe** [comment]
btw where is the second handler for SIGINT?  i just found

```
R_API int r_debug_kill_setup(RDebug *dbg, int sig, int action) {
	eprintf ("TODO: set signal handlers of child\n");
	// TODO: must inject code to call signal()
#if 0
	if (dbg->h->kill_setup)
		return dbg->h->kill_setup (dbg, sig, action);
#endif
	// TODO: implement r_debug_kill_setup
	return false;
}

```

so probably everything should be moved here

**Comment by radare** [comment]
i vote for adding a config var to toggle this behaviour disabled by default for now

**Comment by radare** [comment]
PING

**Pull request merged**